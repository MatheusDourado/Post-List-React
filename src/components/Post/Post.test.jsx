import { render, screen } from "@testing-library/react";
import { Post } from ".";

const props = {
	posts: [
		{
			id: 1,
			title: "title 1",
			body: "body 1",
			cover: "img/img1.png",
		},
		{
			id: 2,
			title: "title 2",
			body: "body 2",
			cover: "img/img2.png",
		},
		{
			id: 3,
			title: "title 3",
			body: "body 3",
			cover: "img/img3.png",
		},
	],
};

describe("<Post />", () => {
	test("should render posts", () => {
		render(<Post {...props} />);

		expect(
			screen.getAllByRole("heading", { name: props.title })
		).toHaveLength(3);
		expect(screen.getAllByRole("img", { name: props.title })).toHaveLength(
			3
		);
		expect(screen.getAllByText(/body/i)).toHaveLength(3);
	});

	test("should not render posts", () => {
		render(<Post />);

		expect(
			screen.queryByRole("heading", { name: props.title })
		).not.toBeInTheDocument();
	});

	test("should match snaphot", () => {
		const { container } = render(<Post {...props} />);

		expect(container.firstChild).toMatchSnapshot();
	});
});
