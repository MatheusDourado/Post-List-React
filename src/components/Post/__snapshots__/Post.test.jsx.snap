// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should match snaphot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      class="post-img"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2
        class="title-1"
      >
        title 1
         
         1
      </h2>
      <p
        class="text"
      >
        body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      class="post-img"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2
        class="title-1"
      >
        title 2
         
         2
      </h2>
      <p
        class="text"
      >
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      class="post-img"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2
        class="title-1"
      >
        title 3
         
         3
      </h2>
      <p
        class="text"
      >
        body 3
      </p>
    </div>
  </div>
</div>
`;
